packages/@n8n/nodes-langchain/nodes/ModelSelector/ModelSelector.node.ts

```diff
@@ -0,0 +1,181 @@
+	description: INodeTypeDescription = {
+		displayName: 'Model Selector',
+		name: 'modelSelector',
+		icon: 'fa:map-signs',
+		defaults: {
+			name: 'Model Selector',
+		},
+		version: 1,
+		group: ['transform'],
+		description:
+			'Use this node to select one of the connected models to this node based on workflow data',
+		inputs: `={{
+				((parameters) => {
+					${configuredInputs.toString()};
+					return configuredInputs(parameters)
+				})($parameter)
+			}}`,
+		outputs: [NodeConnectionTypes.AiLanguageModel],
+		requiredInputs: 1,
+		properties: [
+			numberInputsProperty,
+			{
+				displayName: 'Rules',
+				name: 'rules',
+				placeholder: 'Add Rule',
+				type: 'fixedCollection',
+				typeOptions: {
+					multipleValues: true,
+					sortable: true,
+				},
+				description: 'Rules to map workflow data to specific models',
+				default: {},
+				options: [
+					{
+						displayName: 'Rule',
+						name: 'rule',
+						values: [
+							{
+								displayName: 'Model',
+								name: 'modelIndex',
+								type: 'options',
+								description: 'Choose model input from the list',
+								default: 1,
+								required: true,
+								placeholder: 'Choose model input from the list',
+								typeOptions: {
+									loadOptionsMethod: 'getModels',
+								},
+							},
+							{
+								displayName: 'Conditions',
+								name: 'conditions',
+								placeholder: 'Add Condition',
+								type: 'filter',
+								default: {},
+								typeOptions: {
+									filter: {
+										caseSensitive: true,
+										typeValidation: 'strict',
+										version: 2,
+									},
+								},
+								description: 'Conditions that must be met to select this model',
+							},
+						],
+					},
+				],
+			},
+		],
+	};
+
+	methods = {
+		loadOptions: {
+			async getModels(this: ILoadOptionsFunctions) {
+				const numberInputs = this.getCurrentNodeParameter('numberInputs') as number;
+
+				return Array.from({ length: numberInputs ?? 2 }, (_, i) => ({
+					value: i + 1,
+					name: `Model ${(i + 1).toString()}`,
+				}));
+			},
+		},
+	};
+
+	async supplyData(this: ISupplyDataFunctions, itemIndex: number): Promise<SupplyData> {
+		const models = (await this.getInputConnectionData(
+			NodeConnectionTypes.AiLanguageModel,
+			itemIndex,
+		)) as unknown[];
+
+		if (!models || models.length === 0) {
+			throw new NodeOperationError(this.getNode(), 'No models connected', {
+				itemIndex,
+				description: 'No models found in input connections',
+			});
+		}
+		models.reverse();
+
+		const rules = this.getNodeParameter('rules.rule', itemIndex, []) as any[];
+
+		if (!rules || rules.length === 0) {
+			throw new NodeOperationError(this.getNode(), 'No rules defined', {
+				itemIndex,
+				description: 'At least one rule must be defined to select a model',
+			});
+		}
+
+		for (let i = 0; i < rules.length; i++) {
+			const rule = rules[i] as {
+				modelIndex: string;
+			};
+			const modelIndex = parseInt(rule.modelIndex, 10);
```

Comment:
Does `modelIndex` really come as a string? On my instance I see it as number, so we might not even need to parseInt it.